### ISSUE TITLE
npm plugin fails on launchpad

### ISSUE THREAD

**Issue by ogra1** [canonical/craft-parts#844]
### Bug Description

since several weeks builds with the npm plugin fail in launchpad when an override-build is used and the `craftctl default` step is executed while working fine locally ...

seemingly it fails on this line where it is calling the "npm --version" command:

https://github.com/canonical/craft-parts/blob/main/craft_parts/plugins/npm_plugin.py#L301

since this happens even before the line after the `craftctl default` is executed there must be some internal thing going wrong. please note that this code used to build reliable about two months ago and a simple re-build now reliably fails.

The source as a fork of the customers code is here:

https://github.com/ogra1/everest-demo/tree/main

### To Reproduce

building locally in an lxd container with --destructive mode as well as building by just calling `snapcraft` and leaving the container handling to snapcraft works well ... as soon as the same code from github is attached to an autobuild on launchpad it fails, probably due to a PATH variable setting missing while extracting the node tarball or some such.

### part yaml

```shell
parts:
  nodered:
    plugin: npm
    source: node-red
    npm-include-node: true
    npm-node-version: 22.1.0
    override-build: |
      craftctl default
      npm install
      mkdir -p $CRAFT_PART_INSTALL/lib
      cp -av node_modules $CRAFT_PART_INSTALL/lib/
      ln -s ../lib/node_modules/node-red/red.js $CRAFT_PART_INSTALL/bin/node-red
    prime:
      - -share
```

### Relevant log output
Pull step:

```shell
Pulling nodered
[12/Sep/2024:13:17:02 +0000] "CONNECT nodejs.org:443 HTTP/1.0" 200 16353 "-" "-"
:: + '[' '!' -f /build/everest-demo/parts/nodered/cache/node-v22.1.0-linux-x64.tar.gz ']'
:: + mkdir -p /build/everest-demo/parts/nodered/cache
:: + curl --retry 5 -s https://nodejs.org/dist/v22.1.0/SHASUMS256.txt -o /build/everest-demo/parts/nodered/cache/SHASUMS256.txt
[12/Sep/2024:13:17:02 +0000] "CONNECT nodejs.org:443 HTTP/1.1" 200 10232 "-" "curl/7.81.0"
:: + curl --retry 5 -s https://nodejs.org/dist/v22.1.0/node-v22.1.0-linux-x64.tar.gz -o /build/everest-demo/parts/nodered/cache/node-v22.1.0-linux-x64.tar.gz
[12/Sep/2024:13:17:03 +0000] "CONNECT nodejs.org:443 HTTP/1.1" 200 50739980 "-" "curl/7.81.0"
:: + cd /build/everest-demo/parts/nodered/cache
:: + sha256sum --ignore-missing --strict -c SHASUMS256.txt
:: node-v22.1.0-linux-x64.tar.gz: OK
Pulling edm
```

```shell
building nodered
:: + craftctl default
:: ++ npm --version
:: /build/everest-demo/parts/nodered/run/build.sh: line 5: npm: command not found
:: + NPM_VERSION=
:: error:
:: Traceback (most recent call last):
::   File "/snap/snapcraft/current/lib/python3.10/site-packages/craft_parts/ctl.py", line 111, in main
::     ret = CraftCtl.run(cmd, args)
::   File "/snap/snapcraft/current/lib/python3.10/site-packages/craft_parts/ctl.py", line 45, in run
::     _client(cmd, args)
::   File "/snap/snapcraft/current/lib/python3.10/site-packages/craft_parts/ctl.py", line 98, in _client
::     raise RuntimeError(message)
:: RuntimeError: Failed to run the build script for part 'nodered'.
'override-build' in part 'nodered' failed with code 1.
```
```

**Comment by tigarmo** [comment]
In order to fix this for Snapcraft 8.3.x we need to hotfix into craft-parts 1.33.x

**Comment by tigarmo** [comment]
Fixed in #845
